packages/hyperion-autologging/src/ALElementInfo.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+ */
+
+'use strict';
+
+import { ComponentNameValidator, defaultComponentNameValidator, getReactComponentData_THIS_CAN_BREAK } from './ALReactUtils';
+import type { ReactComponentData } from './ALReactUtils';
+import { getVirtualPropertyValue, setVirtualPropertyValue } from '@hyperion/hyperion-core/src/intercept';
+
+const AL_ELEMENT_INFO_PROPNAME = '__alInfo';
+const AUTO_LOGGING_COMPONENT_TYPE = 'data-auto-logging-component-type';
+
+export default class ALElementInfo {
+  element: Element;
+  _reactComponentData: ReactComponentData | null = null;
+  _reactComponentType: string | null = null;
+  _componentNameValidator: ComponentNameValidator = defaultComponentNameValidator;
+
+  constructor(
+    element: Element,
+    componentNameValidator?: ComponentNameValidator,
+  ) {
+    this.element = element;
+    setVirtualPropertyValue(
+      element,
+      AL_ELEMENT_INFO_PROPNAME,
+      this,
+    );
+
+    this._componentNameValidator = componentNameValidator ?? this._componentNameValidator;
+    this._cacheInfo();
+  }
+
+  _cacheInfo(): void {
+    this.getReactComponentData();
+    this.getReactComponentType();
+  }
+
+  static get(element: Element): ALElementInfo | undefined {
+    return getVirtualPropertyValue<ALElementInfo>(
+      element,
+      AL_ELEMENT_INFO_PROPNAME,
+    );
+  }
+
+  static getOrCreate(element: Element, componentNameValidator?: ComponentNameValidator): ALElementInfo {
+    return ALElementInfo.get(element) ?? new ALElementInfo(element, componentNameValidator);
+  }
+
+  getReactComponentData(): ReactComponentData | null {
+    if (!this._reactComponentData) {
+      this._reactComponentData = getReactComponentData_THIS_CAN_BREAK(
+        this.element,
+        this._componentNameValidator,
+      );
+    }
+    return this._reactComponentData;
+  }
+
+  getReactComponentType(): string | null {
```

Comment:
are we using this anymore? I think in WWW that was left over from the previous implementation.